src/VisualStudio/Core/Def/Implementation/UnusedReferences/Dialog/RemoveUnusedReferencesDialog.xaml.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Windows;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Microsoft.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.UnusedReferences.Dialog
+{
+    /// <summary>
+    /// Interaction logic for RemoveUnusedReferencesDialog.xaml
+    /// </summary>
+    internal partial class RemoveUnusedReferencesDialog : DialogWindow
+    {
+        public string RemoveUnusedReferences => ServicesVSResources.Remove_Unused_References;
+        public string HelpText => ServicesVSResources.Choose_which_action_you_would_like_to_perform_on_the_unused_references;
+        public string Apply => ServicesVSResources.Apply;
+        public string Cancel => ServicesVSResources.Cancel;
+
+        private readonly UnusedReferencesTableProvider _tableProvider;
+
+        public RemoveUnusedReferencesDialog(UnusedReferencesTableProvider tableProvider)
+            : base()
+        {
+            _tableProvider = tableProvider;
+
+            InitializeComponent();
+        }
+
+        public bool? ShowModal(Project project, ImmutableArray<ReferenceUpdate> referenceUpdates)
+        {
+            _tableProvider.AddTableData(project, referenceUpdates);
+
+            using var tableControl = _tableProvider.CreateTableControl();
+            TablePanel.Child = tableControl.Control;
+
+            // The table control updates its view of the datasource on a background thread.
+            // This will force the table to update.
+            ThreadHelper.JoinableTaskFactory.Run(tableControl.ForceUpdateAsync);
+
+            var result = ShowModal();
+
+            TablePanel.Child = null;
+
+            _tableProvider.ClearTableData();
```

Comment:
Consider putting this in a `finally` block to ensure it clears out no matter what if there's an exception; also a comment that this is necessary to prevent leaks might be good.